### ISSUE TITLE
Don't create FlumeViewLevel instance, if indexes.length === 0

### ISSUE THREAD

**Issue by regular** [flumedb/flumeview-query#3]
In a test, I tried to

```
var Flume = require('flumedb')
var MemLog = require('flumelog-memory')
var FlumeQuery = require('flumeview-query')
var pull = require('pull-stream')

var indexes = []

var db = Flume(MemLog())
  .use('query', FlumeQuery(indexes))

db.append({foo: 1}, function (err, seq) {
  console.log(err, seq)
  pull(
    db.query.read({
      $filter: {foo: 1}
    }),
    pull.log()
  )
})
```

 it fails with

```
Error: flumeview-level can only be used with a log that provides a directory
    at create (/Users/regular/dev/ssb/npm-ssb/node_modules/flumeview-level/index.js:22:15)
    at /Users/regular/dev/ssb/npm-ssb/node_modules/flumeview-level/index.js:13:14
    at /Users/regular/dev/ssb/npm-ssb/node_modules/flumeview-query/index.js:35:17
```
My understanding: FlumeViewLevel needs the log to have a file property so that it knows where to put its db for the index. The rest of the code seems to be fine if `indexes=[]`, so not creating FlumeViewLevel in this case should enable combining flumelog-memory with flumeview-query, right? This would be nice for unit testing. Another path could be leveldb in memory.

If this makes sense to you, I could work on a PR. Any thoughts?

**Comment by dominictarr** [comment]
hmm, yeah, if you don't have any indexes it will fallback to a full scan for everything, which may be okay for testing and will still be fast if it's only a small amount of data in memory.

for the case where you have intentionally specified zero indexes this seems reasonable!

:+1:

**Comment by regular** [comment]
PR #4

**Comment by regular** [comment]
fixed in 3.0.4
